### PR TITLE
Update ubuntu latest docs

### PR DESCRIPTION
**Pull request by baumanj**
Signed-off-by: Jon Bauman <5906042+baumanj@users.noreply.github.com>

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: BUILDING.md [hunk 1]
```diff
@@ -78,30 +78,69 @@ echo 'export PKG_CONFIG_PATH="/usr/local/opt/libarchive/lib/pkgconfig:/usr/local
...
 ```
 cp components/hab/install.sh /tmp/
 sh support/linux/install_dev_0_ubuntu_latest.sh
 sh support/linux/install_dev_9_linux.sh
-. ~/.profile
+```
+
+Ubuntu-Server 17.10 has an issue that prevents update from working because it
+tries to read from a CD-ROM that's almost certainly not there. If you get an
+error like this:
```
Comment (fnichol) [inline BUILDING.md]: Woah, that's a bad user experience from Canonical--assuming that they prep these Docker images

**Comment by fnichol** [comment]
Once we're green, looks good to merge

File: BUILDING.md [hunk 2]
```diff
@@ -78,30 +78,69 @@ echo 'export PKG_CONFIG_PATH="/usr/local/opt/libarchive/lib/pkgconfig:/usr/local
...
+
+Compile the project:
+```
 make
 ```
 
-These docs were tested with Ubuntu 16.04 and 16.10 VMs.
+The binaries will be in `habitat/target/debug` and can be run directly, or you can use `cargo` commands in the various `components` subdirectories. For example:
+```
+cd components/sup
```
Comment (elliott-davis) [inline BUILDING.md]: I don't think this is true for the supervisor. It complains about the sup only being able to be run from the launcher. We might want to slap on some additional info about `HAB_SUP_BINARY`

**Comment by fnichol** [comment]
@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/334828884).

File: BUILDING.md [hunk 3]
```diff
@@ -78,30 +78,69 @@ echo 'export PKG_CONFIG_PATH="/usr/local/opt/libarchive/lib/pkgconfig:/usr/local
...
+Then, run the system preparation scripts
 
 ```
 cp components/hab/install.sh /tmp/
 sh support/linux/install_dev_0_ubuntu_latest.sh
 sh support/linux/install_dev_9_linux.sh
-. ~/.profile
+```
+
+Ubuntu-Server 17.10 has an issue that prevents update from working because it
```
Comment (elliott-davis) [inline BUILDING.md]: I think this doc should cover the happy path of installing and building habitat. If this is 100% of the time going to be an issue we should link to the bug tracker for it in a troubleshooting section.

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/334828884).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

File: BUILDING.md (hunk 1 above)
Comment (baumanj) [reply BUILDING.md]: It's not from docker, I'm running the ISO in VMWare. And yeah, it surprised me quite a bit.

File: BUILDING.md (hunk 2 above)
Comment (baumanj) [reply BUILDING.md]: I would if I knew what to write

File: BUILDING.md (hunk 3 above)
Comment (baumanj) [reply BUILDING.md]: I can't find any entry for this in a bug tracker. I haven't tried 17.10.1 yet, so maybe it's fixed there. In any case, I think this works ok for now, and if it's fixed later, I'll update the required versions and remove the workaround.

File: BUILDING.md (hunk 2 above)
Comment (baumanj) [reply BUILDING.md]: I figured out what to write. See https://github.com/habitat-sh/habitat/pull/4516

**Pull request merged**